skin/src/main/java/org/researchstack/skin/DataProviderErrorReceiver.java

```diff
@@ -0,0 +1,128 @@
+package org.researchstack.skin;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Color;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+import android.support.design.widget.Snackbar;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.LocalBroadcastManager;
+import android.view.View;
+import android.widget.TextView;
+
+import com.google.common.base.Strings;
+
+import org.researchstack.backbone.DataProvider;
+import org.researchstack.backbone.onboarding.OnboardingTaskType;
+import org.researchstack.backbone.utils.LogExt;
+import org.researchstack.skin.ui.BaseActivity;
+
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Created by liujoshua on 11/22/2017.
+ */
+public class DataProviderErrorReceiver extends BroadcastReceiver {
+
+    public static final String ACTION_DATA_PROVIDER_ERROR = "org.researchstack.skin.DataProviderError";
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        LogExt.i(getClass(), "Received intent with action: " + intent.getAction());
+
+        if (AppPrefs.getInstance().skippedOnboarding()) {
+            // We don't want to bother a user that has skipped sign-up with the signed out
+            // or consent messages. Short-circuiting until we have an approved message to show
+            // a user that has skipped.
+            return;
+        }
+
+        if (!(context instanceof BaseActivity)) {
+            LogExt.w(getClass(), "Received by class that is not a BaseActivity");
```

Comment:
The only Activity that we really use that extends BaseActivity is MainActivity, and even that is only used to show activities and dashboard data.  Maybe consider instanceof PinCodeActivity to have a broader reach?

Reply:
Good point, PinCodeActivity is the right place for this